homework 2

Marta7799 opened the pull request.

Marta7799 committed:
in progress

KamilMr commented on utils/validators/validator.js:
```diff
@@ -0,0 +1,15 @@
+const joi = require("joi");
+
+const contactSchema = joi.object({
+  name: joi.string().min(3),
```
I think min should be 2 character length (because name Ed exist). And can we add max, for example 50?

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,94 @@
+    });
 
-const updateContact = async (contactId, body) => {}
+    return newContacts;
```
Chyba niepotrzebnie zwracasz contacts tutaj? Bo potem go nie używasz, prawda?

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,66 @@
-})
+router.get("/", async (req, res, next) => {
+  const contacts = await listContacts();
+  console.log("GET /", contacts);
```
Chyba ta linijka jest niepotrzebna gdyż zwracasz w linii 16 kontakty

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,66 @@
+  if (error) return res.status(400).json({ message: error.details[0].message });
+  const contact = await addContact(req.body);
+  if (contact) {
+    res.status(200).json(contact);
```
Status powinien być 201?

KamilMr commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,66 @@
+  if (contact) {
+    res.status(200).json(contact);
+  } else {
+    res.status(400).json({ message: "missing required name field" });
```
Widzę że w linii 30 obsługujesz już sytuację kiedy brakuje jakiś fields więc to nie jest potrzebne. A jeżeli w linii 31 nie sprawdzasz czy pole jest obence to trzeba by tam to dodać.

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,94 @@
+const addContact = async (body) => {
+  try {
+    const { name, email, phone } = body;
+    if (!name || !email || !phone) {
```
To powinno być sprawdzane przez walidator, jest to możliwe?

KamilMr commented on utils/validators/validator.js:
```diff
@@ -0,0 +1,15 @@
+const contactSchema = joi.object({
+  name: joi.string().min(3),
+  email: joi.string().email(),
+  phone: joi.string().min(5),
```
What is someone types in phone input for example 'Hello'? Is is possible to accept number that comes as a string?

KamilMr commented on utils/validators/validator.js:
```diff
@@ -0,0 +1,15 @@
+const joi = require("joi");
+
+const contactSchema = joi.object({
```
Czy sprawdzałaś jak ta schema zachowuje się w sytuacjach kiedy ktoś chciałby umieścić w body jakieś dodatkowe klucze?

KamilMr reviewed (changes requested):
Hej, 
Przeglądnij proszę komentarze które Ci zostawiłem. Potraktuj je jako wstępne sprawdzenie. Kiedy wprowadzisz zmiany daj mi znać i zrobimy kolejne sprawdzenie. :-) W razie pisz śmiało.

Marta7799 committed:
changes

usunięte zwracanie newContacts;
zmienione router.post, router.put;
zmieniony validator
